File: fib.cpp
#include "stdio.h"

int fib(int n) {
    if (n < 2) {
        return n;
    } else return fib(n-2)+fib(n-1);
}

int main() {
    for (int i=2; i<=40;i++) {
        printf("fib(%d)=%d\n", i, fib(i));
    }
    return 0;
}

File: fib.rs
fn main() {
   for L in 2..40 {
     println!("fib({}) = {}",L, fib(L));
   }
}

fn fib(k :i32) -> i32 {
   match k {
     0 => 0,
     1 => 1,
     n => fib(n-2) + fib(n-1)
   }
}
